org.idempiere.test/src/org/idempiere/test/base/POTest.java

```diff
@@ -449,4 +451,13 @@ public void testOptimisticLocking()
 		msg2 = new MMessage(Env.getCtx(), msg1.getAD_Message_ID(), getTrxName());
 		assertEquals(msg1.getMsgText(), msg2.getMsgText());
 	}
+
+	@Test
+	public void testVirtualColumnLoad() {
+		MTest testPo = new MTest(Env.getCtx(), getClass().getName(), 1);
+		testPo.save();
+		BigDecimal expected = new BigDecimal(123.45d).setScale(2, RoundingMode.HALF_DOWN);
+		assertTrue(expected.compareTo(testPo.getTestVirtualQty()) == 0);

```

Comment:
assertEqual is a better choice here. With assertEqual, when it fail, the error message will shown you what's the expected value and what's the actual value.

Reply:
Except that there's no support for `BigDecimal`. In fact, we've changed our original code based on `BigDecimal.doubleValue()` just to keep consistency with other tests we've found (not a thorough analysis though). Maybe this should be addressed for all similar test cases in a separate ticket, perhaps involving Hamcrest. Let me know if you think otherwise.

Reply:
It does works for BigDecimal, it fall back to the standard Object.equals method (i.e assertEqual(Object, Object) which works for BigDecimal too.

Reply:
No, that's not how you'd want to compare two BDs, ever. Please see https://github.com/junit-team/junit4/issues/95

Reply:
It doesn't sound as severe as you make it looks like. As long as the two BD is having the same precision, it work just fine.